plugins/bazaar/src/components/SortView/SortView.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Content,
+  ContentHeader,
+  SupportButton,
+  Progress,
+} from '@backstage/core-components';
+import { AddProjectDialog } from '../AddProjectDialog';
+import { AlertBanner } from '../AlertBanner';
+import { ProjectPreview } from '../ProjectPreview/ProjectPreview';
+import { Button, makeStyles, Link } from '@material-ui/core';
+import { useAsync } from 'react-use';
+import {
+  Entity,
+  EntityRef,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import {
+  catalogApiRef,
+  CATALOG_FILTER_EXISTS,
+} from '@backstage/plugin-catalog-react';
+import { BazaarProject } from '../../types';
+import { bazaarApiRef } from '../../api';
+
+const useStyles = makeStyles({
+  container: {
+    marginTop: '2rem',
+  },
+});
+
+export const SortView = () => {
+  const classes = useStyles();
+  const [openAdd, setOpenAdd] = useState(false);
+  const [openNoProjects, setOpenNoProjects] = useState(false);
+  const [catalogEntities, setCatalogEntities] = useState<Entity[]>([]);
+  const [bazaarMembers, setBazaarMembers] = useState<Map<EntityRef, number>>(
+    new Map(),
+  );
+  const [bazaarProjects, setBazaarProjects] = useState<BazaarProject[]>([]);
+  const bazaarApi = useApi(bazaarApiRef);
+  const catalogApi = useApi(catalogApiRef);
+
+  const compareProjectsByDate = (
+    a: BazaarProject,
+    b: BazaarProject,
+  ): number => {
+    const dateA = new Date(a.updatedAt).getTime();
+    const dateB = new Date(b.updatedAt).getTime();
+    return dateB - dateA;
+  };
+
+  const handleCloseNoProjects = () => {
+    setOpenNoProjects(false);
+  };
+
+  const { loading } = useAsync(async () => {
```

Comment:
probably want to get `loading, error` out of this one to be able to show to the user if there was an error loading the required data